OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -459,26 +459,17 @@ void UpdateCellBlocking(CPos cell)
 			using (new PerfSample("locomotor_cache"))
 			{
 				var cache = blockingCache[cell.Layer];
-
-				var actors = actorMap.GetActorsAt(cell);
 				var cellFlag = CellFlag.HasFreeSpace;
-
-				if (!actors.Any())
-				{
-					cache[cell] = new CellCache(default, cellFlag);
-					return;
-				}
+				var cellImmovablePlayers = default(LongBitSet<PlayerBitMask>);
+				var cellCrushablePlayers = world.AllPlayersMask;
 
 				if (sharesCell && actorMap.HasFreeSubCell(cell))
 				{
-					cache[cell] = new CellCache(default, cellFlag);
+					cache[cell] = new CellCache(cellImmovablePlayers, cellFlag, cellCrushablePlayers);
```

Comment:
This change from `default(LongBitSet<PlayerBitMask>)` to `world.AllPlayersMask` for the third parameter is intentional as well?

Reply:
Yes. Although it doesn't matter here as such. `cellFlag = CellFlag.HasFreeSpace` is in effect, that's checked first. If true then checks for crushable never occur anyway.

But it's cleaner to just set `var cellCrushablePlayers = world.AllPlayersMask;` once at the top. Conceptually it also reflects what we want: You can crush every player until we detect an actor you can't crush.

`cellImmovablePlayers` is the inverse - you can't shove the units out the way until we determine that you can.